resolver.go

```diff
@@ -533,6 +533,7 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 			}
 			if resp.Answer == nil || answers == 0 {
 				logrus.Debugf("[resolver] external DNS %s:%s did not return any %s records for %q", proto, extDNS.IPStr, queryType, name)
+				continue
```

Comment:
Left a comment on your initial PR on moby/moby but I'm wondering if this is correct. If I'm reading correctly, we'd only get here if the DNS would have the domain registered (i.e. didn't return a `NXDOMAIN` and after that responded with an empty response. That feels like am incorrectly configured DNS server (i.e., it should respond with `NXDOMAIN` to indicatie ot doesnt have the domain registered?)

Reply:
In my tests, resp.Rcode returns dns.RcodeSuccess, and resp.Answer == nil || answers == 0, and then returns it to the client. The following are the modified effects(I added a log to print the values of the above three variables):
May 15 14:36:35 selisten-node1 dockerd: time="2019-05-15T14:36:35.465781245+08:00" level=debug msg="Name To resolve: foo-new.test.corp."
May 15 14:36:35 selisten-node1 dockerd: time="2019-05-15T14:36:35.465956811+08:00" level=debug msg="[resolver] query foo-new.test.corp. (A) from 172.1.2.2:35498, forwarding to udp:10.100.146.47"
May 15 14:36:35 selisten-node1 dockerd: time="2019-05-15T14:36:35.467629966+08:00" level=debug msg="[resolver] resp.Answer:[]  answers:0  resp.Rcode:3"
May 15 14:36:35 selisten-node1 dockerd: time="2019-05-15T14:36:35.467663569+08:00" level=debug msg="[resolver] external DNS udp:10.100.146.47 did not return any A records for \"foo-new.test.corp.\""
May 15 14:36:35 selisten-node1 dockerd: time="2019-05-15T14:36:35.467759383+08:00" level=debug msg="[resolver] query foo-new.test.corp. (A) from 172.1.2.2:58000, forwarding to udp:10.142.122.19"
May 15 14:36:35 selisten-node1 dockerd: time="2019-05-15T14:36:35.470913168+08:00" level=debug msg="[resolver] received A record \"10.100.140.12\" for \"foo-new.test.corp.\" from udp:10.142.122.19"

Reply:
So, why does it return a "success", if it doesn't know about the domain?

Reply:
We should probably check https://tools.ietf.org/html/rfc2308 to verify what the correct handling of these is

Reply:
Sorry, the version of docker on my side is 18.06.1-ce. I found that https://github.com/docker/libnetwork/commit/a86d2765b829fb122c70eea7a914d59a8fb1df4a has fixed this bug. Thank you for your reply.

Reply:
Ah! Perfect 👍